### PR TITLE
Interval comparison tests

### PR DESCRIPTION
**Pull request by omus**
A new test suite which performs a variety of comparisons on the various possible kinds of intervals. I found this to be rather useful for catching corner cases as I modified the code.

I've also modified `isless` and `<` to be different. The behaviour of `isless` which is used for sorting just checks the left-endpoint. When the left-endpoints are equal the two intervals are considered equal for sorting and are produced in the order in which they appear.

### CODE REVIEW

File: src/anchoredinterval.jl [hunk 1]
```diff
@@ -1,3 +1,7 @@
+abstract type Anchor end
+struct Ending <: Anchor end
+struct Beginning <: Anchor end
```
Comment (omus) [inline src/anchoredinterval.jl]: I think I can re-use `Direction` here. I wrote this code before endpoints were introduced.

File: src/anchoredinterval.jl [hunk 2]
```diff
@@ -100,6 +104,30 @@ function Base.convert(::Type{Interval{T}}, interval::AnchoredInterval{P, T}) whe
...
+    AnchoredInterval{P, T}(anchor, inclusivity(interval))
+end
+
+function Base.convert(::Type{AnchoredInterval{Ending}}, interval::Interval{T}) where {T}
+    AnchoredInterval{-span(interval), T}(last(interval), inclusivity(interval))
+end
+
+function Base.convert(::Type{AnchoredInterval{Beginning}}, interval::Interval{T}) where {T}
+    AnchoredInterval{span(interval), T}(first(interval), inclusivity(interval))
+end
```
Comment (omus) [inline src/anchoredinterval.jl]: I need to do some strange things to say I want an `AnchoredInterval{<0}`...

File: src/anchoredinterval.jl [hunk 3]
```diff
@@ -100,6 +104,30 @@ function Base.convert(::Type{Interval{T}}, interval::AnchoredInterval{P, T}) whe
...
+    @assert abs(P) == span(interval)
+    anchor = P < zero(P) ? last(interval) : first(interval)
+    AnchoredInterval{P, T}(last(interval), inclusivity(interval))
+end
+
+function Base.convert(::Type{AnchoredInterval{P}}, interval::Interval{T}) where {P, T}
+    @assert abs(P) == span(interval)
+    anchor = P < zero(P) ? last(interval) : first(interval)
+    AnchoredInterval{P, T}(anchor, inclusivity(interval))
+end
```
Comment (omus) [inline src/anchoredinterval.jl]: I'll comment these out as they are currently unused and untested

Comment (omus) [reply src/anchoredinterval.jl]: Done

**Comment by omus** [comment]
AppVeyor 32-bit failure on Julia 0.6 is unrelated (WinRPM download failure). [Code coverage looks good](https://codecov.io/gh/invenia/Intervals.jl/pull/8?src=pr&el=h1)

**Comment by omus** [comment]
Re-running AppVeyor fixed the CI

**Comment by spurll** [comment]
Unless there's something critical in here, I'm planning to hold off on reviewing this stuff for now.

**Comment by spurll** [comment]
I'd be happy to take a look at this next week, but it has some conflicts with the changes we made in #13. If you have time to resolve those let me know, but I don't see this as a particularly pressing PR.

**Comment by omus** [comment]
I'll fix the conflicts
